### PR TITLE
Animate the like button

### PR DESCRIPTION
**Pull request by gaearon**
Inspired by https://github.com/bluesky-social/social-app/pull/4405 but a slightly different animation (inspired by the one in examples on the React homepage).

https://github.com/user-attachments/assets/5d72dfd3-310e-4062-aaf1-7bea8986a28d

https://github.com/user-attachments/assets/a895de5f-66e9-4de9-83dd-aa3be4f7410c

https://github.com/user-attachments/assets/380c7704-3be1-4b5b-8379-1f411e09c623

### CODE REVIEW

**Review by haileyok — APPROVED** [review]
Nice, looks great!

File: src/view/com/util/post-ctrls/PostCtrls.tsx [hunk 1]
```diff
@@ -45,6 +54,7 @@ import {
   Heart2_Stroke2_Corner0_Rounded as HeartIconOutline,
 } from '#/components/icons/Heart2'
 import * as Prompt from '#/components/Prompt'
+import {PlatformInfo} from '../../../../../modules/expo-bluesky-swiss-army'
```
Comment (haileyok) [inline src/view/com/util/post-ctrls/PostCtrls.tsx:57]:
```suggestion
import {PlatformInfo} from '../../../../../modules/expo-bluesky-swiss-army'
```

**Comment by surfdude29** [comment]
Just tried this out on the preview branch, it looks so good! Great job 👌🙏

File: src/view/com/util/post-ctrls/PostCtrls.tsx [hunk 2]
```diff
@@ -104,9 +114,19 @@ let PostCtrls = ({
...
     try {
       if (!post.viewer?.like) {
+        if (PlatformInfo.getIsReducedMotionEnabled()) {
+          likeAnimValue.value = 1
+        } else {
+          likeAnimValue.value = withTiming(1, {
+            duration: 400,
+            easing: Easing.out(Easing.cubic),
+          })
+        }
```
[the remark this answers is not in the document]
Comment (gaearon) [reply src/view/com/util/post-ctrls/PostCtrls.tsx:142]: didn't work :(

**Review by mozzius — APPROVED** [review]
Only briefly had a look on web but this is amazing, 10/10

**Pull request merged**